source/winVersion.py

```diff
@@ -15,6 +16,103 @@
 		winVersionText+=".%d"%winVersion.service_pack_minor
 winVersionText+=" %s" % ("workstation","domain controller","server")[winVersion.product_type-1]
 
+
+@functools.total_ordering
+class WinVersion(object):
+	"""
+	Represents a Windows release.
+	Includes version major, minor, build, service pack information,
+	as well as tools such as checking for specific Windows 10 releases.
+	"""
+
+	def __init__(
+			self,
+			release: str = None,
+			major: int = 0,
+			minor: int = 0,
+			build: int = 0,
+			servicePack: str = ""
+	):
+		if release in (None, ""):
+			self.major = major
+			self.minor = minor
+			self.build = build
+		elif release == "7":
+			self.major = 6
+			self.minor = 1
+			self.build = 7601
+		elif release == "8":
+			self.major = 6
+			self.minor = 2
+			self.build = 9200
+		elif release == "8.1":
+			self.major = 6
+			self.minor = 3
+			self.build = 9600
+		elif release in ("10", "1507"):
+			self.major = 10
+			self.minor = 0
+			self.build = 10240
+		elif release in WIN10_VERSIONS_TO_BUILDS:
+			self.major = 10
+			self.minor = 0
+			self.build = WIN10_VERSIONS_TO_BUILDS[release]
+		else:
+			raise ValueError("Cannot create Windows version information for the specified release")
+		self.servicePack = "1" if release == "7" else ""

```

Comment:
Hi,

I agree with the latter point - using a pool of Windows versions (constants) not only saves us from defining a function to handle complex nature of version checks, it allows easy maintenance in the future in that it will let future contributors define a WinVersion instance representing a specific release. It also makes any part of nVDA calling winVersion module easier to read (app module handler is a key example where we check for Windows 7 versus 8 in some places and these use platform version tuple).

Thanks.

Reply:
Hi,

Also, I'll go ahead and add WinVersion instances representing Windows 7 (SP1) and 8.x for sake of completeness.

Thanks.